exchanges/oex/oex.go

```diff
@@ -0,0 +1,501 @@
+package oex
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wshandler"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+// Oex is the overarching type across this package
+type Oex struct {
+	exchange.Base
+	WebsocketConn *wshandler.WebsocketConnection
+}
+
+const (
+	oexAPIURL     = "https://openapi.oex.com"
+	oexAPIVersion = ""
+
+	// Public endpoints
+	oexGetTicker           = "/open/api/get_ticker"
+	oexGetAllTicker        = "/open/api/get_allticker"
+	oexGetKline            = "/open/api/get_records"
+	oexGetTrades           = "/open/api/get_trades"
+	oexGetLatestCurrPrices = "/open/api/market"
+	oexGetMarketDepth      = "/open/api/market_dept"
+	oexGetAllPairs         = "/open/api/common/symbols"
+	// Authenticated endpoints
+	oexUserInfo        = "/open/api/user/account"
+	oexUserBalanceInfo = "/open/api/user_balance_info"
+	oexAllOrders       = "/open/api/v2/all_order"
+	oexOrderHistory    = "/open/api/all_trade"
+	oexRemoveOrder     = "/open/api/cancel_order"
+	oexRemoveAllOrder  = "/open/api/cancel_order_all"
+	oexCreateOrder     = "/open/api/create_order"
+	oexAllOpenOrders   = "/open/api/v2/new_order"
+	oexSelfTrade       = "/open/api/self_trade"
+	oexFetchOrderInfo  = "/open/api/order_info"
+	oexNoError         = "0"
+)
+
+// SetDefaults sets the basic defaults for Oex
+func (o *Oex) SetDefaults() {
+	o.Name = "Oex"
+	o.Enabled = false
+	o.Verbose = false
+	o.RESTPollingDelay = 10
+	o.RequestCurrencyPairFormat.Delimiter = ""
+	o.RequestCurrencyPairFormat.Uppercase = false
+	o.ConfigCurrencyPairFormat.Delimiter = ""
```

Comment:
Looking at the list of available symbols, some curreny pairs have individual currencies which exceed 3 characters so having a delimiter like "-" is required